### ISSUE TITLE
cant npm install the pacakge

### ISSUE THREAD

**Issue by paaber** [magiclen/node-crc#5]
[2020-05-09T23_33_52_466Z-debug.log](https://github.com/magiclen/node-crc/files/4604894/2020-05-09T23_33_52_466Z-debug.log)
 thats the logfile

**Comment by Nau077** [comment]
+++

**Comment by magiclen** [comment]
@Nau077 Could you provide your log?

**Comment by victorabarros** [comment]
> @Nau077 Could you provide your log?

```bash
victor@ubuntu-samsung:~/victorabarros/Learning/pix-charge-generator$ docker run -it node bash
root@c3c61f692f33:/# mkdir crc
root@c3c61f692f33:/# cd crc/
root@c3c61f692f33:/crc# npm init --force
npm WARN using --force Recommended protections disabled.
Wrote to /crc/package.json:

{
  "name": "crc",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}


npm notice 
npm notice New minor version of npm available! 7.6.3 -> 7.20.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.20.1
npm notice Run npm install -g npm@7.20.1 to update!
npm notice 
root@c3c61f692f33:/crc# ls
package.json
root@c3c61f692f33:/crc# npm install node-crc
npm notice 
npm notice New minor version of npm available! 7.6.3 -> 7.20.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.20.1
npm notice Run npm install -g npm@7.20.1 to update!
npm notice 
npm ERR! code 1
npm ERR! path /crc/node_modules/node-crc
npm ERR! command failed
npm ERR! command sh -c npm run build
npm ERR! > node-crc@2.0.11 build
npm ERR! > cargo-cp-artifact -nc index.node -- cargo build --release --message-format=json-render-diagnostics && tsc
npm ERR! Error: could not find the `cargo` executable.
npm ERR! 
npm ERR! You can find instructions for installing Rust and Cargo at:
npm ERR! 
npm ERR!     https://www.rust-lang.org/tools/install
npm ERR! 
npm ERR! 
npm ERR! Did not copy "cdylib:node-crc"
npm ERR! npm ERR! code 1
npm ERR! npm ERR! path /crc/node_modules/node-crc
npm ERR! npm ERR! command failed
npm ERR! npm ERR! command sh -c cargo-cp-artifact -nc index.node -- cargo build --release --message-format=json-render-diagnostics && tsc
npm ERR! 
npm ERR! npm ERR! A complete log of this run can be found in:
npm ERR! npm ERR!     /root/.npm/_logs/2021-07-27T03_41_16_537Z-debug.log

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-07-27T03_41_16_559Z-debug.log
root@c3c61f692f33:/crc# cat ../root/.npm/_logs/2021-07-27T03_41_16_537Z-debug.log 
0 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
1 info using npm@7.6.3
2 info using node@v15.12.0
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 1ms
5 timing config:load:builtin Completed in 1ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/crc/node_modules/node-crc/.npmrc Completed in 0ms
9 timing config:load:project Completed in 0ms
10 timing config:load:file:/root/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/usr/local/etc/npmrc Completed in 1ms
13 timing config:load:global Completed in 1ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 0ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 5ms
19 verbose npm-session 319661a540924924
20 timing npm:load Completed in 10ms
21 timing command:run Completed in 36ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
22 verbose stack     at ChildProcess.emit (node:events:369:20)
22 verbose stack     at maybeClose (node:internal/child_process:1067:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
23 verbose pkgid node-crc@2.0.11
24 verbose cwd /crc/node_modules/node-crc
25 verbose Linux 5.8.0-63-generic
26 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
27 verbose node v15.12.0
28 verbose npm  v7.6.3
29 error code 1
30 error path /crc/node_modules/node-crc
31 error command failed
32 error command sh -c cargo-cp-artifact -nc index.node -- cargo build --release --message-format=json-render-diagnostics && tsc
33 verbose exit 1
root@c3c61f692f33:/crc# cat ../root/.npm/_logs/2021-07-27T03_41_16_559Z-debug.log 
0 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'install', 'node-crc' ]
1 info using npm@7.6.3
2 info using node@v15.12.0
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 0ms
5 timing config:load:builtin Completed in 0ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/crc/.npmrc Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/root/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 4ms
19 verbose npm-session 1abe27b9e3fdc552
20 timing npm:load Completed in 10ms
21 http fetch GET 304 https://registry.npmjs.org/npm 186ms (from cache)
22 timing arborist:ctor Completed in 0ms
23 timing arborist:ctor Completed in 0ms
24 timing idealTree:init Completed in 7ms
25 timing idealTree:userRequests Completed in 1ms
26 silly idealTree buildDeps
27 silly fetch manifest node-crc@*
28 http fetch GET 200 https://registry.npmjs.org/node-crc 53ms
29 silly placeDep ROOT node-crc@2.0.11 OK for: crc@1.0.0 want: *
30 silly fetch manifest @types/node@^16.4.3
31 silly fetch manifest cargo-cp-artifact@^0.1
32 http fetch GET 200 https://registry.npmjs.org/cargo-cp-artifact 110ms
33 http fetch GET 200 https://registry.npmjs.org/@types%2fnode 151ms
34 timing idealTree:#root Completed in 218ms
35 silly placeDep ROOT @types/node@16.4.3 OK for: node-crc@2.0.11 want: ^16.4.3
36 silly placeDep ROOT cargo-cp-artifact@0.1.4 OK for: node-crc@2.0.11 want: ^0.1
37 timing idealTree:node_modules/node-crc Completed in 10ms
38 timing idealTree:node_modules/@types/node Completed in 0ms
39 timing idealTree:node_modules/cargo-cp-artifact Completed in 0ms
40 timing idealTree:buildDeps Completed in 230ms
41 timing idealTree:fixDepFlags Completed in 0ms
42 timing idealTree Completed in 238ms
43 timing reify:loadTrees Completed in 238ms
44 timing reify:diffTrees Completed in 1ms
45 silly reify moves {}
46 timing reify:retireShallow Completed in 0ms
47 timing reify:createSparse Completed in 1ms
48 timing reify:loadBundles Completed in 0ms
49 silly tarball no local data for cargo-cp-artifact@https://registry.npmjs.org/cargo-cp-artifact/-/cargo-cp-artifact-0.1.4.tgz. Extracting by manifest.
50 silly tarball no local data for node-crc@https://registry.npmjs.org/node-crc/-/node-crc-2.0.11.tgz. Extracting by manifest.
51 silly tarball no local data for @types/node@https://registry.npmjs.org/@types/node/-/node-16.4.3.tgz. Extracting by manifest.
52 http fetch GET 200 https://registry.npmjs.org/cargo-cp-artifact/-/cargo-cp-artifact-0.1.4.tgz 71ms
53 timing reifyNode:node_modules/cargo-cp-artifact Completed in 78ms
54 http fetch GET 200 https://registry.npmjs.org/node-crc/-/node-crc-2.0.11.tgz 132ms
55 timing reifyNode:node_modules/node-crc Completed in 144ms
56 http fetch GET 200 https://registry.npmjs.org/@types/node/-/node-16.4.3.tgz 157ms
57 timing reifyNode:node_modules/@types/node Completed in 173ms
58 timing reify:unpack Completed in 173ms
59 timing reify:unretire Completed in 0ms
60 timing build:queue Completed in 1ms
61 timing build:link:node_modules/cargo-cp-artifact Completed in 1ms
62 timing build:link Completed in 1ms
63 info run node-crc@2.0.11 install node_modules/node-crc npm run build
64 info run node-crc@2.0.11 install { code: 1, signal: null }
65 timing reify:rollback:createSparse Completed in 9ms
66 timing reify:rollback:retireShallow Completed in 0ms
67 timing command:install Completed in 646ms
68 notice 
68 notice New minor version of npm available! 7.6.3 -> 7.20.1
68 notice Changelog: https://github.com/npm/cli/releases/tag/v7.20.1
68 notice Run npm install -g npm@7.20.1 to update!
68 notice 
69 verbose stack Error: command failed
69 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
69 verbose stack     at ChildProcess.emit (node:events:369:20)
69 verbose stack     at maybeClose (node:internal/child_process:1067:16)
69 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
70 verbose pkgid node-crc@2.0.11
71 verbose cwd /crc
72 verbose Linux 5.8.0-63-generic
73 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "node-crc"
74 verbose node v15.12.0
75 verbose npm  v7.6.3
76 error code 1
77 error path /crc/node_modules/node-crc
78 error command failed
79 error command sh -c npm run build
80 error > node-crc@2.0.11 build
80 error > cargo-cp-artifact -nc index.node -- cargo build --release --message-format=json-render-diagnostics && tsc
81 error Error: could not find the `cargo` executable.
81 error
81 error You can find instructions for installing Rust and Cargo at:
81 error
81 error     https://www.rust-lang.org/tools/install
81 error
81 error
81 error Did not copy "cdylib:node-crc"
81 error npm ERR! code 1
81 error npm ERR! path /crc/node_modules/node-crc
81 error npm ERR! command failed
81 error npm ERR! command sh -c cargo-cp-artifact -nc index.node -- cargo build --release --message-format=json-render-diagnostics && tsc
81 error
81 error npm ERR! A complete log of this run can be found in:
81 error npm ERR!     /root/.npm/_logs/2021-07-27T03_41_16_537Z-debug.log
82 verbose exit 1
root@c3c61f692f33:/crc# 
```

**Comment by magiclen** [comment]
> npm ERR! Error: could not find the `cargo` executable.
> npm ERR! 
> npm ERR! You can find instructions for installing Rust and Cargo at:
> npm ERR! 
> npm ERR!     https://www.rust-lang.org/tools/install

Obviously, you didn't set up a Rust development environment. See https://rustup.rs/
